Source/Utils/WarpXAlgorithmSelection.H

```diff
@@ -114,6 +114,16 @@ struct ParticleBoundaryType {
     };
 };
 
+/** MPI reductions
+ */
+struct ReductionType {
+    enum {
+        Maximum = 0,
+        Minimum = 1,
```

Comment:
Is there a specific reason for not using an `enum class` here?
Something like `enum ReductionType {Maximum, Minimum, Sum}`...
You could refer to these constants as `ReductionType::Maximum` and you could use them as template parameters `template <ReductionType RedType>`

Reply:
Hmm, I'm not sure why it's done this way.

Here, I've just reused the same `struct` as for other algorithm selection options, but I agree with you that it looks like we could replace all those structs containing only an `enum` by just enums.

Reply:
Alternatively, for the `int` annotated in the implementation, we could also use the enums from AMReX.